packages/cli-repl/src/completer.ts

```diff
@@ -0,0 +1,76 @@
+import { types as shellTypes } from 'mongosh-shell-api';
+/**
+ * Return complete suggestions given currently typed line
+ *
+ * @param {any} output - The output.
+ *
+ * @returns {string} The output.
+ */
+function completer(line: string): any {
+  // keep initial line param intact to always return in return statement
+  // check for contents of line with:
+  const splitLine = line.split('.');
+  const firstLineEl = splitLine[0];
+  const elToComplete = splitLine[splitLine.length-1];
+
+  const shellComplete = Object.keys(shellTypes.ShellApi.attributes);
+  const collComplete = Object.keys(shellTypes.Collection.attributes);
+  const aggCursorComplete = Object.keys(shellTypes.AggregationCursor.attributes);
+  const collCursorComplete = Object.keys(shellTypes.Cursor.attributes);
+  const rsComplete = Object.keys(shellTypes.ReplicaSet.attributes);
+  const shardComplete = Object.keys(shellTypes.Shard.attributes);
+
+  // suggest SHELLAPI commands
+  if (splitLine.length <= 1) {
+    // TODO: this should also explicitly suggest 'sh', 'rs', and 'db' strings
+    const hits = filterComplete(shellComplete, elToComplete);
+    return [hits.length ? hits : [], line];
+  } else if (firstLineEl === 'db' && splitLine.length === 2) {
+    // TODO: @lrlna suggest DATABASE commands (currently not available in
```

Comment:
Should we create a task for this?

Reply:
i think there is an epic Max created. Shell API only has crud so far, and this is technically not part of crud.